app/models/post.rb

```diff
@@ -0,0 +1,12 @@
+class Post < ApplicationRecord
+  belongs_to :author, class_name: 'User', foreign_key: :author_id
+  has_many :comments
+  has_many :likes
+
+  def self.update_posts_counter(user_id)
+    user = User.find(user_id)
+    user.update(posts_counter: user.posts.count)
+  end
```

Comment:
- [ ] [OPTIONAL] To make your work even better, you can use one of the [Rails Active Callbacks](https://guides.rubyonrails.org/active_record_callbacks.html) - after_save to call the #update_posts_counter method every time a new post is saved to database instead of calling it manually each time, then make the `#update_posts_counter` method private.